### ISSUE TITLE
Constraints in Scrolling API 

### ISSUE THREAD

**Issue by sergey-morenets** [spring-projects/spring-data-jpa#2987]
Hi

We tried to use new feature added in 3.1 - Scrolling API but couldn't understand some of its limitations.

So first of all we created new query method in our repository:

```
public interface OrderRepository extends 
			JpaRepository<Order, Integer> {
	
	Window<Order> findAll(ScrollPosition position);
```

And received an exception at startup:

`Caused by: org.springframework.data.mapping.PropertyReferenceException: No property 'findAll' found for type 'Order'`

The documentation states: _"Scrolling consists of a stable sort, a scroll type (Offset- or Keyset-based scrolling) and result limiting"._
So assuming that method returns Window (or WindowIterator) the error may be **more descriptive** and specify the requirement about sort field in the query name.
Then we added sortable field:

Window<Order> findAllOrderById(ScrollPosition position);

But received another exception:

```
Caused by: java.lang.IllegalStateException: Method public abstract org.springframework.data.domain.Window repository.OrderRepository.findAllOrderById(org.springframework.data.domain.ScrollPosition) expects at least 1 arguments but only found 0; This leaves an operator of type SIMPLE_PROPERTY for property id unbound
	at org.springframework.data.jpa.repository.query.PartTreeJpaQuery.throwExceptionOnArgumentMismatch(PartTreeJpaQuery.java:162)
```

And the error message is not clear for us. Documentation states: "_You can define simple sorting expressions by using property names and define static result limiting using the Top or First keyword through query derivation"._

So adding another field resolves the issue:

`	Window<Order> findByCreatedAtOrderById(LocalDate createdAt, ScrollPosition position);`

However documentation states: "_Offset scrolling uses similar to pagination, an Offset counter to skip a number of results and let the data source only return results beginning at the given Offset"_

But pagination doesn't require us to specify some property for filtering. So the question is why do we need to use filtering property in scrolling-based queries?

**Comment by sergey-morenets** [comment]
We also tried to use @Query annotation:

```
	@Query("SELECT o FROM Order o ORDER BY o.createdAt")
	Window<Order> findAll(ScrollPosition position);
```

But received an exception:

`Caused by: org.springframework.data.repository.query.QueryCreationException: Could not create query for public abstract org.springframework.data.domain.Window repository.OrderRepository.findAll(org.springframework.data.domain.ScrollPosition); Reason: Scroll queries are not supported using String-based queries`

We couldn't find this limitation in the reference documentation so it'd be nice to mention that in the docs.

**Comment by mp911de** [comment]
> However this query still raises an exception:
> 
> ` Window<Order> findAllOrderById(ScrollPosition position);`

This is by design as you do not provide an argument for `Id`. Changing the method declaration to `Window<Order> findAllOrderById(int id, ScrollPosition position);` renders a valid query method.

**Comment by sergey-morenets** [comment]
Yes, that's fine. I again forgot "by" clause so correct method name is:

`	Window<Order> findAllByOrderById(ScrollPosition position);`

**Comment by mp911de** [comment]
> 	Window<Order> findAllByOrderById(ScrollPosition position);

> And "id" argument is not needed here.

Agree after the fix from https://github.com/spring-projects/spring-data-jpa/issues/2987#issuecomment-1569749338.

Is there anything left to do? Otherwise, I would close this ticket.
